python/selfie-lib/selfie_lib/WriteTracker.py

```diff
@@ -197,4 +197,50 @@ def record(
                 )
 
     def persist_writes(self, layout: SnapshotFileLayout):
-        raise NotImplementedError("InlineWriteTracker does not support persist_writes")
+        # Sorting writes based on file name and line number
+        sorted_writes = sorted(
+            self.writes.values(),
+            key=lambda x: (x.call_stack.location.file_name, x.call_stack.location.line),
+        )
+
+        current_file = None
+        content = None
+        delta_line_numbers = 0
```

Comment:
It looks like this code works. But making `current_file` and `content` have a nullable type makes the code below a lot harder to maintain.

```
currentFile: Union[None, String]
content: Union[None, SourceFile]
```

Looking at them as a pair there are four states to consider: (None, None), (None, SourceFile), (String, None), (String, SourceFile).

If you can shift the loop so that they start out as non-null, then there's only one state to consider, (String, SourceFile)

https://github.com/diffplug/selfie/blob/f480d5b99dc9cd7bf36127fd37f5334ebea9001e/jvm/selfie-lib/src/commonMain/kotlin/com/diffplug/selfie/guts/WriteTracker.kt#L194-L213